docs(vue): show complete code context in the "Your First App" tutorial

soundproofboot opened the pull request:
resolves #4165 

# Changes 
- Update code blocks in first 5 pages and page 7 of Vue Photo App tutorial to show additional context for each step with comments to indicate new lines.
- Small changes to surrounding text for clarity.
- Note added linking to localStorage and IndexedDB.

thetaPC reviewed (changes requested):
I noticed that some method names from the Angular version and Vue version do not match even though they have the same code. For example, `addNewToGallery` from Angular and `usePhotoGallery` from Vue. Let's also sync those to match Angular.

The same goes for the `<title>`. Let's match it to Angular.

Lastly, some of the content also doesn't match each other. A good example, Angular Build Your First App mentions:

```
We put the visual aspects of our app into <ion-content>. In this case, it’s where we’ll add a button that opens the device’s camera as well as displays the image captured by the camera. Start by adding a [floating action button](https://ionicframework.com/docs/api/fab) (FAB) to the bottom of the page and set the camera image as the icon.
```

While Vue states:
```
We put the visual aspects of our app into <ion-content>. In this case, it’s where we’ll add a button that opens the device’s camera as well as displays the image captured by the camera. But first, remove both the ExploreContainer component and its import statement:
```
It should match the steps of Angular.

soundproofboot commented:
@thetaPC Hi Maria, I tried to make the titles match in Vue and added some `<head>` content that was in the Angular pages but not Vue. I changed the name of the `takePhoto` method to `addNewGallery` to match the Angular method that is comparable. One thing worth noting is these tutorials link to [repos](https://github.com/ionic-team/tutorial-photo-gallery-vue/blob/main/src/composables/usePhotoGallery.ts) so I was trying to not change the code too much and leave the documentation not matching the repo.

On making the steps match Angular, there are more steps required for Vue in that specific case so I think they are slightly reordered to clarify what's being removed and added in Vue (as in remove these two things before we add these other several things) vs. Angular, where I was able to clearly indicate what to remove and add in one code block. I have not looked at this in a while so it may take me a minute to familiarize myself with the frameworks and changes but I'm willing to keep working on this if there are other suggestions.

thetaPC commented:
@soundproofboot thank you so much! To ensure that all 'Your First App' PRs are merged more quickly, I will be assisting to push them to completion. The main obstacle is synchronizing all three frameworks. Please bear with me as I make the necessary changes.

brandyscarney commented on docs/vue/your-first-app.md:
```diff
@@ -89,45 +88,61 @@ npm install @capacitor/camera @capacitor/preferences @capacitor/filesystem
+// CHANGE: Add the following import
 import { defineCustomElements } from '@ionic/pwa-elements/loader';
+
+/* Ionic styles are not shown in this example to keep it brief but will be included in the Ionic package downloaded for your app. Do not remove them. */
```
This does not follow the same flow as the Angular example. Should it be like the following:

```suggestion
/* ...existing Ionic styles... */
```

Or should we just include them?

thetaPC replied:
https://github.com/ionic-team/ionic-docs/pull/4197/commits/233761afa2b2790f12aea40055a4ff48a306921b and https://github.com/ionic-team/ionic-docs/pull/4197/commits/cef88ca09302d250291f4186dde515d7d2b450ef

brandyscarney commented on docs/vue/your-first-app.md:
````diff
@@ -146,30 +161,53 @@ Open `/src/views/Tab2.vue`. We see:
 
-```html
+```vue
 <ion-title>Photo Gallery</ion-title>
````
This may be confusing if people interpret it to change the large title to this:

```diff
-<ion-title size="large">Tab 2</ion-title>
+<ion-title>Photo Gallery</ion-title>
```

I realize this is the same as the Angular example, just pointing it out because I didn't notice it until this review.

thetaPC replied:
Made the change for Angular: https://github.com/ionic-team/ionic-docs/pull/4157/commits/c19dbea64e542db250c18deeb74c1b76e707da06

thetaPC replied:
https://github.com/ionic-team/ionic-docs/pull/4197/commits/5ba929275907d3b20cdc9ec9c514feba65d3bf84

brandyscarney commented on docs/vue/your-first-app/3-saving-photos.md:
```diff
@@ -1,70 +1,228 @@
+    photos.value = [savedImageFile, ...photos.value];
+  };
+
+  // CHANGE: Add `savePicture()` method
```
This was added in the previous step so it isn't a change in this step.

```suggestion

```

brandyscarney reviewed (commented):
Requested some changes on v8 that will also apply to v7. 🙂

thetaPC commented on docs/vue/your-first-app/4-loading-photos.md:
```diff
@@ -1,69 +1,323 @@
-watch(photos, cachePhotos);
+    // CHANGE: Display the photo by reading into base64 format
+    for (const photo of photosInPreferences) {
+      const file = await Filesystem.file({
```
https://github.com/ionic-team/ionic-docs/pull/4197/commits/98c520d7e1bce34c8e21a6d26a7014e319596963

thetaPC commented on docs/vue/your-first-app/4-loading-photos.md:
```diff
@@ -1,69 +1,323 @@
+    const photosInPreferences = photoList.value ? JSON.parse(photoList.value) : [];
+
+    for (const photo of photosInPreferences) {
+      const file = await Filesystem.file({
```
https://github.com/ionic-team/ionic-docs/pull/4197/commits/98c520d7e1bce34c8e21a6d26a7014e319596963

thetaPC commented on docs/vue/your-first-app/7-live-reload.md:
```diff
@@ -86,49 +134,28 @@ const showActionSheet = async (photo: UserPhoto) => {
+      <ion-img :src="photo.webviewPath" @click="showActionSheet(photo)"></ion-img>
+    </ion-col>
+  </ion-row>
+</ion-grid>
```
I was trying to keep it consistent with the other frameworks. For example, Angular has it's code in [two different files.](https://github.com/ionic-team/ionic-docs/pull/4157/files#diff-0af3eb2d2322ccc994877acf333cdb6080930493d0b7fd49e11b41b7f514c565R131) So I wanted to separate it out with Vue as well.

https://github.com/ionic-team/ionic-docs/pull/4197/commits/81db1c3df047fb8c242a59690e1e016b7a6fdfbb

thetaPC commented on docs/vue/your-first-app/5-adding-mobile.md:
````diff
@@ -30,15 +57,58 @@ const savePicture = async (photo: Photo, fileName: string): Promise<UserPhoto> =
+
+Then update `savePicture()` to look like the following:
+
+```ts
````
Ah, the reason I did this was because the examples with the surrounding code were shown at the beginning of the tutorial. As the developer progresses, I expect them to feel comfortable with what they've done so we don't need to have the surrounding code as they would know where in the file is the update function. Thoughts?

brandyscarney commented on docs/vue/your-first-app.md:
The CLI sets it automatically for all of them (except if `cordova` is passed for Angular), so it looks like we can remove it for Angular and [React](https://github.com/ionic-team/ionic-docs/pull/4201/files?show-viewed-files=true#diff-7aed867b1c952dfd530b15c3e2d6c7f1374b705b82d52b2a333cf738995afa78R74) too:

```ts
  if (this.schema.type === 'react' || this.schema.type === 'vue') {
    options['capacitor'] = true;
  }

  if (
    (this.schema.type === 'angular' || this.schema.type === 'angular-standalone')
    && options['cordova'] === null
  ) {
    options['capacitor'] = true;
  }
```

thetaPC replied:
[Angular](https://github.com/ionic-team/ionic-docs/pull/4157/commits/7d154f8f7803276904f2b778e4e679b1d3d11b75)

brandyscarney commented on docs/vue/your-first-app.md:
```diff
@@ -184,37 +222,45 @@ import {
   IonCol,
   IonImg,
```
Hmmm okay. I think it would be a better user experience if they didn't have errors in their IDE while going through the steps. Maybe we can make a follow-up task for this?

thetaPC replied:
https://github.com/ionic-team/ionic-docs/pull/4197/commits/b78bb35c665ed5e035fa0ee985163d522b71e660

thetaPC commented on docs/vue/your-first-app/4-loading-photos.md:
```diff
@@ -1,69 +1,323 @@
-watch(photos, cachePhotos);
+    // CHANGE: Display the photo by reading into base64 format
+    for (const photo of photosInPreferences) {
+      const readFile = await Filesystem.file({
```
https://github.com/ionic-team/ionic-docs/pull/4197/commits/eb865feded571682a53dd409e83090df900b7c80

thetaPC commented on docs/vue/your-first-app/5-adding-mobile.md:
````diff
@@ -30,15 +57,58 @@ const savePicture = async (photo: Photo, fileName: string): Promise<UserPhoto> =
+When running on mobile, set `filepath` to the result of the `writeFile()` operation - `savedFile.uri`. When setting the `webviewPath`, use the special `Capacitor.convertFileSrc()` method ([details on the File Protocol](../../core-concepts/webview.md#file-protocol)). To use this method, we'll need to import Capacitor into `usePhotoGallery.ts`.
+
+```ts
+import { Capacitor } from '@capacitor/core';
````
https://github.com/ionic-team/ionic-docs/pull/4197/commits/d082cdcb16241836abe208c59d9f18a391e136e9

thetaPC reviewed (approved):
LGTM
